### DISCUSSION TITLE
Docker ~/.config directory is empty

### DISCUSSION THREAD

**Discussion by khanakia** [coder/code-server#5132] [Q&A]
I am using the docker image for the code server https://coder.com/docs/code-server/latest/install#docker

HOST Machine: `~/.config` is empty

When i ssh into the docker container `/home/coder/.config` is also empty

Actually, it using the `/root/.config/code-server/config.yaml` this directory for config file within the Docker Container

So how to fix this ?

Note if I used this modified command it works
```
docker run -it --name code-server -p 8003:8080 \
  -v "/root/.config:/root/.config" \
  -v "$PWD:/home/coder/project" \
  -u "$(id -u):$(id -g)" \
  -e "DOCKER_USER=$USER" \
  codercom/code-server:latest
```

**Comment by jsjoeio** [comment]
I'm not super familiar with how the `.config` should work in a Docker environment. 

cc @code-asher maybe?

**Comment by code-asher** [comment]
I am able to reproduce if I run the Docker command as root.  In our entry script we attempt to rename the `coder` user to the passed-in `DOCKER_USER` but if this is `root` then I imagine this will fail since `root` already exists?  So maybe what we need to do is if the user is `root` we skip the rename and instead change root's home directory to `/home/coder`.

Or we add a note to the docs to bind to `/root/.config` instead when running as root.

**Reply by code-asher** [reply]
The ideal solution would probably be to keep using the `coder` user in the container instead of root but I am not sure how to do that.

**Reply by khanakia** [reply]
the docker container runs as root actually so it will use the /root/.config file within the docker container, not the /home/coder/.config, and also if we run the container as `coder` user some files we will not be able to edit as the host will still have files with root permission.
